### PR TITLE
sidecar: document caveat of --shipper.upload_compacted

### PR DESCRIPTION
**Pull request by belm0**
Document a known issue of the experimental `--shipper.upload_compacted` flag, where the different compaction schedules of Prometheus and Thanos is not accounted for and may affect higher-order compaction and downsampling.

See #2413 

Signed-off-by: John Belmonte <john@neggie.net>

### CODE REVIEW

File: docs/components/sidecar.md [hunk 1]
```diff
@@ -79,6 +79,9 @@ To use this, the Prometheus compaction needs to be disabled. This can be done by
 - `--storage.tsdb.min-block-duration=2h`
 - `--storage.tsdb.max-block-duration=2h`
 
+A known issue is that Prometheus and Thanos use different compaction schedules, so Compact may
```
Comment (bwplotka) [inline docs/components/sidecar.md:82]: This is nice, can you link issue as well?

However, for documentation would be nice to be clear what we mean by "confused".. until we know roughly what can go wrong this piece of documentation will be only source of further questions.

**Comment by belm0** [comment]
> Nice, some suggestion.
> 
> BTW I also don't think it should be confused (: It actually works as designed. The downsampling will kick at some point where blocks are packed into 2w no ?

As I've observed, the 162h blocks from prometheus do not get further compacted.

When we are mixing compacted blocks from different schedules, it means that blocks at the same compaction level have different sizes, and a block and level N may be bigger than a block with level N+1.  I don't know how the prometheus compaction library behaves in those circumstances, and whether the behavior is even defined.

File: docs/components/sidecar.md (hunk 1 above)
Comment (belm0) [reply docs/components/sidecar.md:82]: A agree the language is vague, but it's because we actually don't know.  See my other comment about not knowing how the compaction library deals with blocks mixed together from different compaction schedules.

I guess that's why `upload_compacted` is considered experimental...

Comment (bwplotka) [reply docs/components/sidecar.md:82]: Well then we should either learn or not describe until we learn.

We have some idea though as you mentioned below

**Comment by bwplotka** [comment]
So TSDB compaction should be smart enough to fill the gap for bigger level compaction if it does not overlap. So there might be a case IMO where.

lv4 = 2w
you have 3x 6d blocks. In this case none of them will be ever compacted.

But BTW. This is totally fine! (: We should not be worried much. Yes it will not downsample, but with you change we should be fine. 

We can add more tests here to know those edge cases: https://github.com/prometheus/prometheus/blob/5225f3ff5047e7c36a2beb1ae0376b788f07086b/tsdb/compact_test.go#L166 

((:

File: docs/components/sidecar.md [hunk 2]
```diff
@@ -79,6 +79,9 @@ To use this, the Prometheus compaction needs to be disabled. This can be done by
...
+get confused when trying to further compact or downsample compacted blocks from Prometheus.
```
Comment (bwplotka) [inline docs/components/sidecar.md:83]:
```suggestion
get confused when trying to further compact or downsample compacted blocks from Prometheus.
This is similar to why enabling local compaction with upload is tricky.

This can be overcome by enabling experimental vertical compaction.
```
